src-local/ca_continuation_utils.py

```
@@ -0,0 +1,647 @@
+    
+    Returns:
+        Adjusted tolerance
+    """
+    if Ca_interval_width > 1e-2:
+        return 1e-3  # Coarse tolerance for wide intervals
+    elif Ca_interval_width > 1e-4:
+        return 1e-4  # Medium tolerance
+    else:
+        return base_tolerance  # Fine tolerance near convergence
+
+def get_adaptive_max_nodes(Ca_interval_width: float, theta_min: Optional[float] = None) -> int:
```

Ruff (0.11.9)

Redefinition of unused `get_adaptive_max_nodes` from line 13

(F811)